--- libcurl/curl/lib/vtls/openssl.c.orig	2022-01-13 19:12:10.183880350 +0100
+++ libcurl/curl/lib/vtls/openssl.c	2022-01-13 19:13:21.131414933 +0100
@@ -3052,7 +3052,7 @@
     }
   }
 
-#if defined(OPENSSL_VERSION_MAJOR) && (OPENSSL_VERSION_MAJOR >= 3)
+#if defined(OPENSSL_VERSION_MAJOR) && (OPENSSL_VERSION_MAJOR >= 3) && NEVERDEFINED
   /* OpenSSL 3.0.0 has deprecated SSL_CTX_load_verify_locations */
   {
     if(ssl_cafile) {
