--- libgdal/gdal/gcore/gdalexif.cpp.orig	2022-11-06 21:09:20.000000000 +0100
+++ libgdal/gdal/gcore/gdalexif.cpp	2022-11-18 16:57:25.681013955 +0100
@@ -459,6 +459,7 @@
 /*                                                                      */
 /*      Extract all entry from a IFD                                    */
 /************************************************************************/
+#pragma optimize("", off)
 CPLErr EXIFExtractMetadata(char**& papszMetadata,
                            void *fpInL, int nOffset,
                            int bSwabflag, int nTIFFHEADER,
@@ -763,6 +764,7 @@
 
     return CE_None;
 }
+#pragma optimize("", on)
 
 /************************************************************************/
 /*                        WriteLEUInt16()                               */
